### DISCUSSION TITLE
http2_adapter 2.2.0 发布失败的问题

### DISCUSSION THREAD

**Discussion by CaiJingLong** [cfug/dio_issue_release_action#23] [General]
这个 [action](https://github.com/cfug/dio/actions/runs/4545799804/jobs/8013621193#step:5:151) 运行失败，因为 pub 发布插件中 dry run 有一个 warning.
<img width="1394" alt="image" src="https://user-images.githubusercontent.com/14145407/228406540-8151058d-4a5e-4b41-a18c-17f8c682ce2f.png">

但因为 action 的 1.0.7版本是先 commit 再 release，所以 https://github.com/cfug/dio/commit/f35f7d529d0c69174b3a24e80d3654aaa25e390b 仍被提交

我在 [1.0.8](https://github.com/cfug/dio_issue_release_action/releases/tag/v1.0.8) 中添加了一个步骤，会先执行 dry-run publish 再提交代码，这样在测试发布时会先尝试发布，如果失败，则不提交代码，以避免回滚

**Comment by CaiJingLong** [comment]
但需要讨论的是，今后 merge 到主干的代码是否应该先确保 lint 能通过
@AlexV525

**Reply by AlexV525** [reply]
这里 lint 没通过有另一个问题，是在 https://github.com/cfug/dio/pull/1719 中有讨论过的需要在根项目配置 pubspec 的问题，目前由于没有配置导致实际上并未检查相关的 lint。我会开一个 PR 补上。

**Comment by AlexV525** [comment]
> 我在 [1.0.8](https://github.com/cfug/dio_issue_release_action/releases/tag/v1.0.8) 中添加了一个步骤，会先执行 dry-run publish 再提交代码，这样在测试发布时会先尝试发布，如果失败，则不提交代码，以避免回滚

还没看代码，不过加入前置 dry run 即可，测试依旧可以跳过
